src/paths/steiner.cpp

```diff
@@ -0,0 +1,480 @@
+					min_F = F;
+				}
+			}
+
+			igraph_vector_int_t vectorlist_1;
+			IGRAPH_CHECK(igraph_vector_int_init(&vectorlist_1,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&vectorlist_1);
+
+			igraph_vector_int_t edgelist_1;
+			IGRAPH_CHECK(igraph_vector_int_init(&edgelist_1,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&edgelist_1);
+
+			igraph_get_shortest_path_dijkstra(graph,&vectorlist_1,&edgelist_1,k,min_E_value,weights,IGRAPH_ALL);
+			
+			
+			igraph_vector_int_append(vectorlist_all,&vectorlist_1);
+			igraph_vector_int_append(edgelist_all,&edgelist_1);
+
+		
+			igraph_vector_int_destroy(&vectorlist_1);
+			igraph_vector_int_destroy(&edgelist_1);
+			IGRAPH_FINALLY_CLEAN(2);
+
+		}
+		else {
+			
+			igraph_integer_t E1,F1;
+
+			E1 = *D.begin();
+			F1 = *next(D.begin(),1);;	
+			
+
+
+			igraph_vector_int_t vectorlist_1;
+			IGRAPH_CHECK(igraph_vector_int_init(&vectorlist_1,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&vectorlist_1);
+
+			igraph_vector_int_t edgelist_1;
+			IGRAPH_CHECK(igraph_vector_int_init(&edgelist_1,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&edgelist_1);
+
+			igraph_get_shortest_path_dijkstra(graph,&vectorlist_1,&edgelist_1,k,E1,weights,IGRAPH_ALL);
+
+			igraph_vector_int_t vectorlist_2;
+			IGRAPH_CHECK(igraph_vector_int_init(&vectorlist_2,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&vectorlist_2);
+
+			igraph_vector_int_t edgelist_2;
+			IGRAPH_CHECK(igraph_vector_int_init(&edgelist_2,1));
+			IGRAPH_FINALLY(igraph_vector_int_destroy,&edgelist_2);
+
+			igraph_get_shortest_path_dijkstra(graph,&vectorlist_2,&edgelist_2,k,F1,weights,IGRAPH_ALL);
+
+			igraph_vector_int_append(vectorlist_all,&vectorlist_1);
+			igraph_vector_int_append(vectorlist_all,&vectorlist_2);
+
+			igraph_vector_int_append(edgelist_all,&edgelist_1);
+			igraph_vector_int_append(edgelist_all,&edgelist_2);
+
+			igraph_vector_int_destroy(&vectorlist_2);
+			igraph_vector_int_destroy(&vectorlist_1);
+			
+			igraph_vector_int_destroy(&edgelist_1);
+			igraph_vector_int_destroy(&edgelist_2);
+
+			IGRAPH_FINALLY_CLEAN(4);
+
+			std::set<igraph_integer_t> min_F;
+			min_F.insert(F1); 
+
+
+
+		}
+		
+		m = k;
+		D = min_F;			
+		
+		igraph_vector_int_destroy(&vectorlist);
+		igraph_vector_int_destroy(&edgelist);
+		IGRAPH_FINALLY_CLEAN(2);
+	}
+	
+
+
+	return IGRAPH_SUCCESS;
+
+}
+
+
+igraph_error_t igraph_steiner_dreyfus_wagner(const igraph_t *graph,const igraph_vector_int_t* steiner_terminals,
+igraph_neimode_t mode, const igraph_vector_t *weights,igraph_real_t *res,igraph_vector_int_t *res_tree)
+{
+	if (mode != IGRAPH_ALL)
+	{
+		*res = 0;
+		IGRAPH_ERROR( "Currently this function only supports undirected graphs while the graph's mode is not undirected.\n",IGRAPH_FAILURE);
+		return IGRAPH_FAILURE;
+	}
+	
+	
+	igraph_integer_t no_of_vertices = igraph_vcount(graph);
+	igraph_integer_t no_of_edges = igraph_ecount(graph);
+
+	
+	if (no_of_vertices == 0 || (no_of_vertices == 1)) //graph is empty
```

Comment:
Can we not generalize this function to handle the 0 and 1-vertex cases in a reasonable way? The spanning tree functions do. Instead of throwing an error, check that arguments are valid (`steiner_terminals` contains only valid vertex IDs, etc.) and return a reasonable result.